packages/modules/Extension-Ads/src/worker/worker.js

```diff
@@ -228,14 +246,18 @@ const extensionAdsAppMgr = {
 
     console.log(`%cReceiving ad from tab ${tabId} - ${tabUrl}, ads detected: ${this.tabData[tabId].ads.size}`, 'color: green; font-weight: bold');
     console.log('Tab data:', this.tabData[tabId]);
+    this.rateAdsIfNeeded(tabId);
     this.sendAdsIfNeeded(tabId);
   },
   async _onMessage_adClicked(data, from) {
-    const { url: tabUrl } = from.tab;
+    const { id: tabId, url: tabUrl } = from.tab;
     const { meta, adData, clickedUrl } = data;
 
     if (clickedUrl) {
+      const trackedAd = this.tabData[tabId].ads.get(clickedUrl);
+
       const result = await this.processAdData(adData, tabUrl, clickedUrl);
+      result.adId = trackedAd ? trackedAd.adId : result.adId;
```

Comment:
This is incorrect, there should be only one place for assigning the Id for advertising.